File: src/utils/logger/logger.go
package logger

import (
	"log"
	"os"
)

type Logger struct {
	info  *log.Logger
	err   *log.Logger
	fatal *log.Logger
}

func NewLogger() *Logger {
	info := log.New(os.Stdout, "INFO: ", log.Ldate|log.Ltime)
	err := log.New(os.Stdout, "ERROR: ", log.Ldate|log.Ltime)
	fatal := log.New(os.Stdout, "FATAL: ", log.Ldate|log.Ltime)

	return &Logger{
		info:  info,
		err:   err,
		fatal: fatal,
	}
}

func (l *Logger) Info(msg string, a ...any) {
	l.info.Printf(msg+"\n", a...)
}

func (l *Logger) Error(msg string, a ...any) {
	l.err.Printf(msg+"\n", a...)
}

func (l *Logger) Fatal(msg string, a ...any) {
	l.fatal.Printf(msg+"\n", a...)
	os.Exit(1)
}

File: src/pyutils/logger.py
import logging
import sys


class Logger:
    def __init__(self) -> None:
        logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s', datefmt='%Y-%m-%d %H:%M:%S')

        self.info = logging.getLogger('info')
        self.error = logging.getLogger('error')
        self.fatal = logging.getLogger('fatal')

    def info(self, msg: str) -> None:
        self.info.info(msg)

    def error(self, msg: str) -> None:
        self.error.error(msg)

    def fatal(self, msg: str) -> None:
        self.fatal.fatal(msg)
        sys.exit(1)